doc/releases/changelog-dev.md

```diff
@@ -13,6 +13,19 @@
 
 <h3>Improvements</h3>
 
+* `DiagonalOperations` is not a separate subclass any more. 
+  [(#1889)](https://github.com/PennyLaneAI/pennylane/pull/1889) 
+
+  Instead, devices can check for the diagonal 
+  property using attributes:
+
+  ``` python
+  from pennylane.ops.qubit.attributes import diagonal_in_z_basis
```

Comment:
very very minor, but would the name `qml.ops.qubit.attributes.diagonal` also suffice?

Reply:
The `commute_contolled` transform actually checks the current `basis` attribute of operations to determine whether they can be pushed through controlled ops. Having this attribute that specifies the z basis is nice because we could in principle also add `diagonal_in_x_basis` (and one for y) and then remove the `basis` attribute altogether from `Operator`